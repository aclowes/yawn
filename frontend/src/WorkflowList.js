import React from 'react';
import {Alert, Table} from 'react-bootstrap';
import {Link} from 'react-router';

import API from "./API";

export default class WorkflowList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {workflows: null};
  }

  componentDidMount() {
    API.get('/api/workflows/', (payload, error) => {
      this.setState({workflows: payload, error});
    })
  }

  renderRows() {
    if (this.state.workflows === null) {
      return (
        <tr>
          <td colSpan="4">Loading...</td>
        </tr>
      )
    } else {
      return this.state.workflows.map((workflow) => (
        <tr key={workflow.id}>
          <td><Link to={`/workflows/${workflow.id}`}>{workflow.name}</Link></td>
          <td>{workflow.schedule_active ? 'True' : 'False'}</td>
          <td>{workflow.schedule || '(none)'}</td>
          <td>{workflow.tasks.length}</td>
        </tr>
      ))
    }
  }

  render() {
    return (
      <div>
        <h3>Workflows</h3>
        {this.state.error && <Alert bsStyle="danger">{this.state.error}</Alert>}
        <Table striped bordered condensed hover>
          <thead>
          <tr>
            <th>Name</th>
            <th>Active</th>
            <th>Schedule</th>
            <th>Tasks</th>
          </tr>
          </thead>
          <tbody>
          {this.renderRows()}
          </tbody>
        </Table>
      </div>
    );
  }
}
