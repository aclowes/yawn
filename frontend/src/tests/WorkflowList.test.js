import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import WorkflowList from '../WorkflowList';
import {mockAPI} from './mocks'

it('WorkflowList loading', () => {
  mockAPI([null]);
  const component = ReactTestRenderer.create(<WorkflowList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('WorkflowList failure', () => {
  mockAPI();
  const component = ReactTestRenderer.create(<WorkflowList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('WorkflowList success', () => {
  mockAPI([[{
    "id": 2,
    "current_version": 1,
    "current_version_id": 2,
    "task_count": 30,
    "schedule": null,
    "schedule_active": false,
    "versions": null,
    "name": "Big Ol' Workflow Example"
  }]]);
  const component = ReactTestRenderer.create(<WorkflowList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
