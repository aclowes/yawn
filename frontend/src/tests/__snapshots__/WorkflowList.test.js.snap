// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowList failure 1`] = `
<div>
  <h3>
    Workflows
  </h3>
  <div
    className="alert alert-danger"
    role="alert"
  >
    some error message
  </div>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Schedule Active
        </th>
        <th>
          Schedule
        </th>
        <th>
          Tasks
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="4"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WorkflowList loading 1`] = `
<div>
  <h3>
    Workflows
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Schedule Active
        </th>
        <th>
          Schedule
        </th>
        <th>
          Tasks
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="4"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WorkflowList success 1`] = `
<div>
  <h3>
    Workflows
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Schedule Active
        </th>
        <th>
          Schedule
        </th>
        <th>
          Tasks
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            Big Ol' Workflow Example
          </a>
        </td>
        <td>
          False
        </td>
        <td>
          (none)
        </td>
        <td>
          30
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
